onnxscript/rewriter/ort_fusions/cos_sin_cache.py

```diff
@@ -47,13 +43,66 @@ def __init__(
         # matched nodes as part of the rewrite-step. We apply a separate final
         # pass to remove unused nodes.
         super().__init__(name, remove_nodes=False)
-        self._max_pos_id = max_pos_id
+        # TODO: Determine what should be the default max_pos_id value
+        self._max_pos_id = 2048  # Set a default max_pos_id value
+        self._max_pos_id_from_model = False
```

Comment:
I think it would be simpler to combine the above two, using a None value for _max_pos_id (to indicate dynamic calculation of max-pos-id).